### PR TITLE
IITC-Mobile: 'Support DeX desktop mode' option

### PR DESCRIPTION
**Pull request by Whomiga**
Useful for automatically making the display to be desktop when running on the Samsung DeX desktop system while allowing for showing in regular mode on a smartphone without having the switch back and forth between Desktop and regular when moving between the two environments.

### CODE REVIEW

File: mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_Mobile.java
Comment (johnd0e) [inline mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_Mobile.java]: The same condition repeats in different parts of code.
Shouldn't it be separate function?

Comment (Whomiga) [reply mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_Mobile.java]: I added the Dex code to where it was already being determined for the original Desktop, if I put it to a separate function should I pull the entire if statement?

Least amount of changes was by adding it to the original Desktop determination, this also allows for the original Desktop determination to override the DeX determination.

I didn't want to break something if I modified that and lost proper scope.

**Comment by Whomiga** [comment]
Somewhere in the functionized change, I lost the DeX mode change...

**Comment by johnd0e** [comment]
@Whomiga 
I have no Samsung to test, so I just ask:
- Is this patch finished?
- Does it completely solve your original issue?
- Is IITC fully usable in DeX-mode (considering #162)?

**Comment by Whomiga** [comment]
This patch, as written, does solve the original issue. Noticed #162 while testing the changes, but I believe the problems with DeX was preexisting, and unconnected to my changes. The IITC problems in DeX Mode will require further testing.

**Comment by johnd0e** [comment]
> I believe the problems with DeX was preexisting, and unconnected to my changes.

Sure, unconnected.
But if IITC in DeX is barely usable than this patch is not needed too.

**Comment by Whomiga** [comment]
I would recommend keeping the patch on the inactive list by cancelling the pull request until more has been determined about the problems with DeX mode. In any case, I have a testable situation here and can continue to test with my branch as necessary.

**Comment by Whomiga** [comment]
I've found that the DeX problems were a Red Herring - the problem is that the draw tools plugin, and maybe other stuff, has trouble being used with a connected mouse and appears to only work correctly when used by touch.

**Comment by Whomiga** [comment]
@johnd0e - If IITC Mobile is used in DeX mode with the Phone acting as a Touchpad, it works perfectly, I believe that this Pull Request is quite valid as it does what it says and only showed an evidently preexisting problem.

**Comment by johnd0e** [comment]
So this PR is ready to merge, right?

**Comment by Whomiga** [comment]
Yes, I see no problem with merging this pull request, if there are no further questions, I'll merge it (If I can, not sure how that goes)

**Comment by Whomiga** [comment]
@johnd0e Question: How do I properly merge this pull request and close it. It has been a while and I don't know if I have Write Access to Merge, etc...

**Comment by johnd0e** [comment]
We will merge it later, with the rest of PRs.

**Comment by Whomiga** [comment]
Any way that I can move this pull request from the master branch to another branch, like "DeX-Updates", so I can do other work while leaving this to be merged later? (Didn't plan ahead)

**Comment by johnd0e** [comment]
Well, you could (by creating new PR), but it'd be totally pointless.
Just leave `master` and create new branch in your fork to continue work.

**Comment by johnd0e** [comment]
@Whomiga
This option needs better name/description.
And do you want it to be enabled by default?

Is it possible to remove it for DeX-less systems?

**Comment by Whomiga** [comment]
The option is totally ignored on DeX-less systems and defaults to not-selected in default settings.

The main feature of this is that it will automatically come up in a Desktop more if the phone is in DeX mode, otherwise it will come up in whatever mode is selected in settings. For a phone, that setting is usually in regular mode due to the display on the phone.

If selected it will 'force' the default behavior of IITC to be displayed in a Desktop mode when the DeX is active (This would be a phone connected to a DeX setup) so that it acts as you would expect on a Desktop environment. If not in a DeX environment or selected, it would default to the normal settings which could be set to a Desktop environment or a Phone environment depending on the user's requirements.

I mostly added this because I use my phone in both a normal environment and a DeX environment and got tired of resetting to/from Desktop every time I switched from Dex to Phone. This way it would come up as a Desktop in the DeX environment and a regular mobile environment outside of DeX.

I figured anyone who uses their phone in both DeX and Mobile environments would be able to use this capability.

> 
> @Whomiga
> This option needs better name/description.
> And do you want it to be enabled by default?

It is disabled by default, as that causes it to use the original setting mode with no respect to the phone even having the DeX mode capabilities.
 
> Is it possible to remove it for DeX-less systems?

See no reason why it is necessary, as the setting value is totally ignored on phones that can't do DeX
(maybe a note in the release indicating that it has no effect except on DeX systems)

**Comment by johnd0e** [comment]
> I figured anyone who uses their phone in both DeX and Mobile environments would be able to use this capability.

Sure. I know it's purpose.


> It is disabled by default

When I open Settings - I see it in enabled state.


> See no reason why it is necessary

Reason: to not confuse users of most other systems: they have no clue what is that "DeX".

**Comment by Whomiga** [comment]
> 
> 
> > I figured anyone who uses their phone in both DeX and Mobile environments would be able to use this capability.
> 
> Sure. I know it's purpose.
> 
> > It is disabled by default
> 
> When I open Settings - I see it in enabled state.

Oops, sorry it appears I was wrong.  But if the phone doesn't have a Samsung DeX mode, it wouldn't matter what the value was set to.

> > See no reason why it is necessary
> 
> Reason: to not confuse users of most other systems: they have no clue what is that "DeX".

You could make it "Samsung DeX" vs "DeX" - if anyone wanted more into they could look it up online or make the name and description...
"Force Samsung DeX desktop mode" 
"Force desktop mode on systems when Samsung DeX mode is active"

I mostly picked the title "Force DeX desktop mode" because of the "Force desktop mode" settings title of the previous settings item.

**Comment by Whomiga** [comment]
I changed the preference to disabled by default, but I really think that anyone who uses Samsung DeX would want the desktop mode by default if they are using the Samsung DeX mode.

File: mobile/app/src/main/res/values/strings.xml
Comment (johnd0e) [inline mobile/app/src/main/res/values/strings.xml]:
```suggestion
    <string name="pref_dex_desktop">Support Samsung DeX Mode</string>
```

File: mobile/app/src/main/res/values/strings.xml
Comment (johnd0e) [inline mobile/app/src/main/res/values/strings.xml]:
```suggestion
    <string name="pref_dex_desktop_sum">Force desktop mode when DeX mode is active</string>
```

**Review by johnd0e — CHANGES_REQUESTED** [review]
How about this wording?

File: mobile/app/src/main/res/values/strings.xml
Comment (Whomiga) [reply mobile/app/src/main/res/values/strings.xml]: To nit-pick:
Technically DeX is supported whether this is selected or not, this setting just forces IITC to always open in Desktop mode when running on an Active DeX system rather than having to toggle Desktop on and off when switching between and Active DeX system and normal mobile phone mode. 

But, this would work.

Comment (Whomiga) [reply mobile/app/src/main/res/values/strings.xml]: I can easily make the change in my copy and submit it. What is the thought of making it on/off by default?  I really think that anyone using a DeX mode - which is the only time this would be of any use would want it on.

**Comment by Whomiga** [comment]
Default on/off is the only remaining question, I think.

**Comment by johnd0e** [comment]
Well, make it 'on' if you wish. 
And should we merge it squashed or maybe you want to make some cleanup in this branch?

**Comment by johnd0e** [comment]
@Whomiga?

**Comment by Whomiga** [comment]
> 
> 
> @Whomiga?

Sorry, been away for a few days...

Do I need to do any cleanup? I believe all the changes are in this pull request to be merged.

File: mobile/app/src/main/res/xml/preferences.xml
Comment (johnd0e) [inline mobile/app/src/main/res/xml/preferences.xml]: So you've decided to keep it disabled

File: mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_Mobile.java
Comment (johnd0e) [inline mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_Mobile.java]: Wrong indentation

**Review by johnd0e — COMMENTED** [review]
@Whomiga

File: mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_Mobile.java
Comment (Whomiga) [reply mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_Mobile.java]: Hmmm... my local copy shows that it is enabled, but diffs don't show it as being different than on the server...

Comment (Whomiga) [reply mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_Mobile.java]: I just committed a new copy that should have a fixed indentation and the item enabled by default, but the change doesn't show in diffs on the server

Comment (Whomiga) [reply mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_Mobile.java]: Looks like I finally got all those changes pushed to the server

**Pull request merged**
